### PR TITLE
Filter transform support service

### LINKED ISSUE #3325: Filter & Transform node-support_service content ## Problem Statement
Our tome-sync data doesn't match up with our templates, and we need to follow the steps [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2835) to ensure that the data will be compatible with our current templates.

## Goal
Content from `node-support_service` entities match their corresponding entities from `pages.json`

### PR DESCRIPTION
**Pull request by bkjohnson**
### Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/3325

This filters & transforms content for `node-support_service` entities.

There is also some refactoring which pulls out some of the common transformations around link-type objects into a helper.  This touches the following entity types in addition to `node-support_service`:
- `paragraph-link_teaser`
- `paragraph-react_widget`
- `taxonomy_term-administration`

## Testing done

The usual

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-support_service.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        fieldLink: {
+          type: ['object', 'null'],
+          properties: {
+            url: {
+              type: 'object',
+              properties: {
+                path: { type: 'string' },
+              },
+              required: ['path'],
+            },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/transformed/node-support_service.js]: Man, I wish we could refactor these while we're here. Glad this makes it easy for somebody else to do later, though.

Wishful thinking. :smile:

File: src/site/stages/build/process-cms-exports/transformers/helpers.js [hunk 2]
```diff
@@ -78,12 +88,28 @@ module.exports = {
...
+   * If this object exists, an object will be returned matching what is expected for "fieldLink" objects,
+   * otherwise null is returned.
+   *
+   * If the optional parameter is used, the returned object will only contain those attributes.
+   *
+   * @param {array} fieldLink
+   * @param {array} attrs
+   * @return {object}
    */
-  uriToUrl(uri) {
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/helpers.js]: Do we need to still export this function?

File: src/site/stages/build/process-cms-exports/transformers/helpers.js [hunk 3]
```diff
@@ -78,12 +88,28 @@ module.exports = {
...
-    return uri;
+  createLink(fieldLink, attrs = ['url', 'title', 'options']) {
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/helpers.js]: This is handy. :+1:

**Review by cvalarida — APPROVED** [review]
Looks like this was a small node! That's great!

File: src/site/stages/build/process-cms-exports/transformers/helpers.js (hunk 2 above)
Comment (bkjohnson) [reply src/site/stages/build/process-cms-exports/transformers/helpers.js]: Nope - with the refactor it is only used for the `createLink` function in the helper file:

image

**Pull request merged**